src/Workspaces/CSharp/Portable/Simplification/Simplifiers/NameSimplifier.cs

```diff
@@ -333,6 +333,13 @@ private NameSimplifier()
                 case SyntaxKind.IdentifierName:
                     identifier = ((IdentifierNameSyntax)name).Identifier;
```

Comment:
move definition of 'identifier' into this case clause (and the one below).  it is not read after teh switch.